### PR TITLE
test: add more teachers test

### PR DESCRIPTION
**Pull request by RickaPrincy**

### CODE REVIEW

File: src/__tests__/Manager.Teacher.cy.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    cy.get('[data-key="ref"]').click()
+    cy.get('#ref').type(teacher1Mock.ref)
+    cy.wait('@getFilters')
+    cy.get('tbody tr').should('have.length', 1).should('not.contain', teachersMock[1].first_name)
+    cy.get('tbody tr').should('have.length', 1)
+    cy.get('tbody tr:first-child').should('contain', teacher1Mock.first_name)
+    unmount()
+  })
+
+  it('can edit teaher (click on "Éditer" button)', () => {
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js]: #### R3. What exactly are you testing here ?
- **EditButton**: routing ?
- **Teacher Edition**: if so, you should do some assertions for the edited teacher after the `@putUpdate` (to ensure that the teacher edition works correctly)

File: src/__tests__/Manager.Teacher.cy.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    cy.get('button').contains('Connexion').click()
+    cy.intercept('GET', '/whoami', whoamiManagerMock).as('getWhoami')
+    cy.intercept('GET', `/managers/${manager1Mock.id}`, manager1Mock).as('getManager')
+    cy.intercept('GET', '/teachers?page=1&page_size=10', teachersMock).as('getTeachers')
+    cy.intercept('GET', /teachers\?page=1&page_size=10&(first_name|ref|last_name)=/, [teacher1Mock]).as('getFilters')
+    cy.intercept('GET', `/teachers/${teachersMock[0].id}`, teachersMock[0]).as('getTeachers1')
+    cy.intercept('PUT', `/teachers`, [updatedInfo]).as('putUpdate')
+
+    cy.wait('@getWhoami')
+    cy.get(':nth-child(2) > .MuiListItem-root').click()
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js]: R1. What is `:nth-child(2) > .MuiListItem-root` here ?

* get element by `data-testid` or add some comments that describe it

File: src/__tests__/Manager.Teacher.cy.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  })
+
+  it('list all teachers', () => {
+    cy.wait('@getManager')
+    cy.viewport(1000, 950)
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js:35]: R2. why are you setting the viewport in this test case only ?

**/!\ set it directly in the cypress conf to apply it to every test case**

File: src/__tests__/Manager.Teacher.cy.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    cy.wait('@getFilters')
+    cy.get('tbody tr').should('have.length', 1).should('not.contain', teachersMock[1].first_name)
+    cy.get('tbody tr').should('have.length', 1)
+    cy.get('tbody tr:first-child').should('contain', teacher1Mock.first_name)
+    unmount()
+  })
+
+  it('can filter teachers by last_name', () => {
+    cy.get('#first_name').clear()
+    cy.get('.css-t6tio9-RaFilterButton-root > .MuiButtonBase-root').click()
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js]: same as **R1**

File: src/__tests__/Manager.Teacher.cy.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    cy.get('.Mui-focusVisible').click()
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js:52]: same as **R1**

File: src/__tests__/Manager.Teacher.cy.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    cy.get('#last_name').type(teacher1Mock.last_name)
+    cy.wait('@getFilters')
+    cy.get('tbody tr').should('have.length', 1).should('not.contain', teachersMock[1].first_name)
+    cy.get('tbody tr').should('have.length', 1)
+    cy.get('tbody tr:first-child').should('contain', teacher1Mock.first_name)
+    unmount()
+  })
+
+  it('can filter teachers by ref', () => {
+    cy.get('.css-t6tio9-RaFilterButton-root > .MuiButtonBase-root').click()
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js]: same as **R1**

File: src/__tests__/Manager.Teacher.cy.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    cy.get('#ref').type(teacher1Mock.ref)
+    cy.wait('@getFilters')
+    cy.get('tbody tr').should('have.length', 1).should('not.contain', teachersMock[1].first_name)
+    cy.get('tbody tr').should('have.length', 1)
+    cy.get('tbody tr:first-child').should('contain', teacher1Mock.first_name)
+    unmount()
+  })
+
+  it('can edit teaher (click on "Éditer" button)', () => {
+    cy.get(':nth-child(1) > .column-undefined > .MuiButtonBase-root').click()
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js]: same as **R1**

File: src/__tests__/Manager.Teacher.cy.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    cy.wait('@getTeachers1')
+    cy.get('#address').type(updatedInfo.address)
+    cy.get('[for="status_DISABLED"]').click()
+    cy.get('[data-testid="SaveIcon"]').click()
+    cy.wait('@putUpdate')
+    unmount()
```
Comment (YumeT023) [inline src/__tests__/Manager.Teacher.cy.js:85]: R4. use the `afterEach()` hook to _unmount()_ after each test case the same way you did for _afterEach()_

**Review by YumeT023 — CHANGES_REQUESTED** [review]

File: src/__tests__/Manager.Teacher.cy.js (hunk 3 above)
Comment (RickaPrincy) [reply src/__tests__/Manager.Teacher.cy.js:35]: The button appears only on > 1000 viewport

File: src/__tests__/Manager.Teacher.cy.js (hunk 6 above)
Comment (RickaPrincy) [reply src/__tests__/Manager.Teacher.cy.js]: done

**Pull request closed without merging**